core/pkg/server/resume.go

```diff
@@ -73,126 +100,132 @@ func (r *ResumeState) Update(
 				Code:    service.ErrorInfo_USAGE,
 			}}
 		return result, err
+	default:
+		if err := r.update(bucket, run, config); err != nil && r.resume == Must {
+			result := &service.RunUpdateResult{
+				Error: &service.ErrorInfo{
+					Message: err.Error(),
+					Code:    service.ErrorInfo_UNKNOWN,
+				},
+			}
+			return result, err
+		}
+		run.Resumed = true
```

Comment:
Should this still be set to `true` if `err != nil`?

Reply:
I don’t think so, since we will return an error back to init and won’t actually resume